src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,257 @@
+                using var _ = ArrayBuilder<TextDocumentEdit>.GetInstance(out var textDocumentEdits);
+                foreach (ApplyChangesOperation applyChangesOperation in applyChangesOperations)
+                {
+                    var solution = document.Project.Solution;
+                    var changes = applyChangesOperation.ChangedSolution.GetChanges(solution);
+                    var projectChanges = changes.GetProjectChanges();
+
+                    // TO-DO: If the change involves adding a document, execute via command instead of WorkspaceEdit
+                    // until adding documents is supported in LSP: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After support is added, remove the below if-statement and add code to support adding documents.
+                    var addedDocuments = projectChanges.SelectMany(
+                        pc => pc.GetAddedDocuments().Concat(pc.GetAddedAdditionalDocuments().Concat(pc.GetAddedAnalyzerConfigDocuments())));
+                    if (addedDocuments.Any())
+                    {
+                        return Array.Empty<TextDocumentEdit>();
+                    }
+
+                    var changedDocuments = projectChanges.SelectMany(pc => pc.GetChangedDocuments());
+                    var changedAnalyzerConfigDocuments = projectChanges.SelectMany(pc => pc.GetChangedAnalyzerConfigDocuments());
+                    var changedAdditionalDocuments = projectChanges.SelectMany(pc => pc.GetChangedAdditionalDocuments());
+
+                    // TO-DO: If the change involves modifying any document besides the document where the code action
+                    // was invoked, temporarily execute via command instead of WorkspaceEdit until LSP bug is fixed:
+                    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After bug is fixed, remove the below if-statement and the existing code should work.
+                    if (changedDocuments.Any(d => d != document.Id) ||
+                        changedAnalyzerConfigDocuments.Any() ||
+                        changedAdditionalDocuments.Any())
+                    {
+                        return Array.Empty<TextDocumentEdit>();
+                    }
+
+                    // Changed documents
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetDocument, solution.GetDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed analyzer config documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
```

Comment:
this seems wrong.  you're passing in changedDocuments instead of changedAnalyzerConfigDocuments (same issue below).  This feels untested as i would have thought this just wouldn't work.  and/or would crash.